doc/generated_testcases_v2.yaml

```diff
@@ -0,0 +1,110 @@
+# The following extensions will be created for each .in that's found.
+# Any key starting with a dot will be interpreted as an extension generator.
+# .ans will be generated before all others.
+.ans: /solutions/accepted/sol.py
+.png: /visualizers/asy.py # Note that directly calling a shell executable is not allowed
+
+
+# Optionally, a salt for generating the {seed} variables. Will be prepended to
+# the command being run.
+random_salt: abcd
+
+
+## Layout of generators/ directory:
+# - stdout.py <val>:             takes a value and writes it to stdout
+# - stdout.cpp <val>:            idem, but is a c++ file.
+# - dir/lib.h
+# - dir/generator.cpp <val>:     idem, but now includes a header. Works like validators.
+# - lib.py
+# - generator.py <val>:          uses lib.py. Fine because python will be executed locally.
+# - lib.sh
+# - generator.sh <val>:          uses lib.sh. Fine because shell will be executed locally.
+
+
+data:
+  # Introduce the `sample` directory.
+  sample:
+    type: directory
+    .ans: # empty to disable generating .ans files here
+    data:
+      1.in: stdout.py 1  # prints `1` to stdout, which is piped to 1.in
+
+# Manual cases
+
+      2.in: # empty value to indicate a manual case
+      3.in: /manual_cases/sample/3.in # Key that matches `/*.in` glob is a manual testcase that will be copied from elsewhere.
+     #4.in: # not listing a test is OK for development but will give warnings.
+
+  secret:
+    type: directory
+    data:
+
+# Types of generator programs.
+      01.in: stdout.py  3    
+      02.in: stdout.cpp 4   # c++ is compiled as for validators
+      03.in: dir 5          # directories are OK as for validators
+      04.in: generator.py 5 # python files may include local files. They are executed locally.
+      05.in: generator.py 5 # python files may include local files. They are executed locally.
+
+# The entire value is passed through python shlex.shlex.
+      06-string.in: stdout.py "a b" # gives a single string of length 3 as argument
+
+# {seed.*} anywhere in the argument string will be replaced by a 31-bit integer
```

Comment:
Can `{seed.*}` contain spaces or other weird characters? I.e. how would the following be interpreted:
    01.in: stdout.py {seed  < foo > /dev/null #}

Reply:
Is seed an 31-bit unsigned or signed integer?